components/gitpod-db/src/typeorm/typeorm.ts

```diff
@@ -61,19 +62,28 @@ export class TypeORM {
         } as ConnectionOptions;
     }
 
-    public async getConnection() {
+    public async getConnection(): Promise<Connection> {
+        const connectionManager = getConnectionManager();
         if (this._connection === undefined) {
-            const connectionMgr = getConnectionManager();
-            if (connectionMgr.has(TypeORM.DEFAULT_CONNECTION_NAME)) {
-                this._connection = connectionMgr.get(TypeORM.DEFAULT_CONNECTION_NAME);
+            this._connection = new Deferred();
+
+            let connection: Connection;
+            if (connectionManager.has(TypeORM.DEFAULT_CONNECTION_NAME)) {
+                // This path is important for the CLI, where the "default" connection at this point already
+                // is initialized from "ormconfig.ts"
+                connection = connectionManager.get(TypeORM.DEFAULT_CONNECTION_NAME);
             } else {
-                this._connection = await createConnection({
+                // Default path for all apps (server, etc.)
+                connection = await createConnection({
+                    // cannot be connectionMgr.create() as suggested by docs
                     ...this._options,
                     name: TypeORM.DEFAULT_CONNECTION_NAME,
                 });
             }
+            this._connection.resolve(connection);
```

Comment:
@AlexTugarev The difference is in the `Deferred`: Before, it was easy to end up with more then one connection being created:
- 1st caller went into getConnection and got stuck in createConnection (this takes up to 20s!)
- 2nd caller went into getConnection and immediately got the a `Conenction` - just a not fully initialized one

Using `Deferred`, line 86 allows us to block all callers to wait for the initialization to complete.

Reply:
I see, so there are to different ways to obtain the connection used in server. 'Wasn't aware of it. 

Reducing to a single `Deferred` makes sense.

nit: for readability it would be helpful to have separate function, if applicable, something like `getDefaultConnection` in addition.